src/ucp/proto/proto_common.c

```diff
@@ -490,6 +492,15 @@ static ucp_lane_index_t ucp_proto_common_find_lanes_internal(
                 goto out;
             }
 
+            if (flags & UCP_PROTO_COMMON_INIT_FLAG_RKEY_PTR) {
+                cmpt_attr = ucp_cmpt_attr_by_md_index(context, md_index);
```

Comment:
IMO can move this to line 462

Reply:
i meant just the line `cmpt_attr = ucp_cmpt_attr_by_md_index(context, md_index);`
and then use single if with &&

Reply:
ok, done. Left the check outside `UCP_PROTO_COMMON_INIT_FLAG_REMOTE_ACCESS` branch, because it does check anything in rkey_config